utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt

```diff
@@ -0,0 +1,32 @@
+package org.oppia.android.util.profile
+
+import android.content.Intent
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.util.extensions.getProtoExtra
+import org.oppia.android.util.extensions.putProtoExtra
+
+private const val PROFILE_ID_INTENT_DECORATOR = "PROFILE_ID_INTENT_DECORATOR"
+
+/** Decorator for wrapping a user's profile [ProfileId] within its intent. */
+object CurrentUserProfileIdIntentDecorator {
+  /**
+   * Packs the intent with a [ProfileId] proto object that sets it as the current
+   * screen.
+   *
+   * [extractCurrentUserProfileId] should be used for retrieving the screen name later.
+   */
+  fun Intent.decorateWithUserProfileId(profileId: ProfileId) {
+    putProtoExtra(
+      PROFILE_ID_INTENT_DECORATOR,
+      profileId
+    )
+  }
+
+  /** Returns the [ProfileId] packed in the intent via [decorateWithUserProfileId]. */
```

Comment:
Or default instance if there is no profile ID bundled in the intent.

Reply:
Noted, makes sense.